### DISCUSSION TITLE
Refresh data in "overview widget" (BaseWidget) according to selected table filters

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#2552] [Help] [answered]
What I would like to do is use ["overview widget"](https://filamentphp.com/docs/2.x/admin/dashboard#stats-overview-widgets) to display statistics based on selected table filters. 

To make it easier to visualize what's going on: 

1. Table displays tasks with amounts to pay for them.
2. I got filters to query them by date range.
3. I would like widget to show sum of tasks amount according to selected table filters.

For chart widgets there is a **getData()** method, but it doesn`t seems to work with "overview widget". I need to access somehow selected filters / query or records.

The only way I figured out to do was emitting event from **ListSettlements**:
```php
protected function getTableQuery(): Builder
{
  $this->emit('updateSettlementsMonthSummaryWidget', [
    'month' => $this->tableFilters['date_range']['month'] ?? date('m'),
    'year' => $this->tableFilters['date_range']['year'] ?? date('Y'),
  ]);

  return Settlement::query();
}
```

and then in widget listen for it and perform some calculations:
```php
protected $listeners = ['updateSettlementsMonthSummaryWidget' => 'update'];

public array $monthsSum;

public function update($dateRange)
{
  //some logic to calculate $this->monthsSum
}

protected function getCards(): array
{
  return [
    Card::make('Sum for selected date range', number_format($this->monthsSum[0], 2, ',', '.') . ' PLN')
      ->description($this->getDescription())
      ->descriptionIcon($this->getDescriptionIcon())
      ->chart($this->monthsSum)
      ->color($this->getColor()),
  ];
}
```

This works and data refreshes while changing filters, except of chart. Its data is also changing, but it is not reflected. Is there any way to refresh it also?

The second question is, if there is any better way than this to access and refresh data in this type of widget?

**Comment by studiowizjo** [comment, marked as answer]
Solved by: https://github.com/filamentphp/filament/pull/2729

**Reply by invaders-xx** [reply]
@studiowizjo, I followed your trick and I can't make it working with graph widget, the graph is not refreshing

**Reply by studiowizjo** [reply]
Sorry, for late reply. I am now very out of time, but I will check this at weekend. Probably this type of widget need similiar changes to introduced in mentioned above PR.

**Reply by studiowizjo** [reply]
Once again, sorry for late reply. It looks like [further commit](https://github.com/filamentphp/filament/commit/97be9a5b6b224765e93cb144e4ea05e8583bed5d) with refactor break things.

My understanding of things is that this line:

```php
chart = Chart.getChart(this.$refs.canvas);
```

is essential, so Chart.js is able to find own instance. Otherwise if it is pulled into Alpine, for some reason it can`t.

My suggestion is to rollback with eventual method rename as @danharrin did, but besides ```init()``` method should be like previous.

**Reply by danharrin** [reply]
Can you PR your suggestion? I dont understand
